src/lib/icu/Compare.cxx

```diff
@@ -112,3 +112,32 @@ IcuCompare::IsIn(const char *haystack) const noexcept
 	return false;
 #endif
 }
+
+bool
+IcuCompare::StartsWith(const char *haystack) const noexcept
+{
+#ifdef HAVE_ICU_CASE_FOLD
+	return StringIsEqual(IcuCaseFold(haystack).c_str(),
+			  needle.c_str(), strlen(needle.c_str()));
+#elif defined(_WIN32)
+	if (needle == nullptr)
+		/* the MultiByteToWideChar() call in the constructor
+		   has failed, so let's always fail the comparison */
+		return false;
+
+	try {
+		auto w_haystack = MultiByteToWideChar(CP_UTF8, haystack);
+		return FindNLSStringEx(LOCALE_NAME_INVARIANT,
+				       FIND_STARTSWITH|NORM_IGNORECASE,
+				       w_haystack.c_str(), -1,
+				       needle.c_str(), -1,
+				       nullptr,
+				       nullptr, nullptr, 0) == 0;
+	} catch (...) {
+		/* MultiByteToWideChar() has failed */
+		return false;
+	}
+#else
+	return strncmp(haystack, needle.c_str(), strlen(needle.c_str()));
```

Comment:
That check was inverted; I fixed it.
Yeah, the C library is confusing - better use MPD's wrapper functions like `StringStartsWith()`.